### PR TITLE
Add video overlay option to video player

### PR DESCRIPTION
**Pull request by sammorrisdesign-zz**
## What does this change?

Adds an optional field to the `Video` template to add a video overlay to the player. This adds a headline, tag if needed and video duration. At the moment it's not visible to users. But it needs to go in as it will be used in my [Video Container PR](https://github.com/guardian/frontend/pull/12584/) and some stuff by @blongden73.
## Request for comment

@rich-nguyen @jamesgorrie 
## 

_Does this PR meet the [contributing guidelines](https://github.com/guardian/frontend/blob/issue_pr_templates/.github/CONTRIBUTING.md#submission)?_

### CODE REVIEW

File: common/app/model/Element.scala [hunk 1]
```diff
@@ -94,6 +94,11 @@ final case class VideoMedia(videoAssets: List[VideoAsset]) {
     }.flatten.sorted
   }
   val duration: Int = videoAssets.headOption.map(_.duration).getOrElse(0)
+  val formattedDuration: String = {
+    videoAssets.headOption.map(_.durationMinutes).getOrElse(0).toString() + ":" + "%02d".format(videoAssets.headOption.map(_.durationSeconds).getOrElse(0))
+  }
+  val durationMinutes: Int = videoAssets.headOption.map(_.durationMinutes).getOrElse(0)
```
Comment (jamesgorrie) [inline common/app/model/Element.scala]: Should these not be options, as, if the time isn't there, it should be `None` not 0?

File: common/app/model/Asset.scala [hunk 2]
```diff
@@ -107,8 +107,9 @@ case class VideoAsset(
   // The video duration in seconds
   val duration: Int = fields.get("durationSeconds").getOrElse("0").toInt +
                            (fields.get("durationMinutes").getOrElse("0").toInt * 60)
+  val durationSeconds: Int = fields.get("durationSeconds").getOrElse("0").toInt
```
Comment (jamesgorrie) [inline common/app/model/Asset.scala]: On [this point](https://github.com/guardian/frontend/pull/12591/files#r60062191) - Could we not have:
`val durationSeconds: Option[Int] = fields.get("durationSeconds").map(_.toInt)`

File: common/app/views/fragments/media/video.scala.html [hunk 3]
```diff
@@ -67,6 +67,21 @@
                     </object>
                 }
             </video>
+            @if(showOverlay){
+            <div class="video-overlay">
```
Comment (jamesgorrie) [inline common/app/views/fragments/media/video.scala.html]: Is this already styled, or is that done in the other PR?

Comment (sammorrisdesign) [inline common/app/views/fragments/media/video.scala.html]: That's done in a different PR

File: common/app/model/Asset.scala (hunk 2 above)
Comment (sammorrisdesign) [inline common/app/model/Asset.scala]: I think we would still want to display 0:32, if there is no `durationMinutes`

Comment (jamesgorrie) [inline common/app/model/Asset.scala]: We receive `durationMinutes: "0"` from the API - the reason for the change above is to not have the duration set to 0:00 when it isn't a media element.

Comment (jamesgorrie) [inline common/app/model/Asset.scala]: Sorry, I am being a fool, I didn't realise these were video specific.

You could make this slightly more readable by:
```
val durationSeconds: Int = fields.get("durationSeconds").map(_.toInt).getOrElse(0)
val durationMinutes: Int = fields.get("durationMinutes").map(_.toInt).getOrElse(0)
val duration: Int = durationSeconds + (durationMinutes * 60)
```

Saying that - why do we need `duration` in seconds?

**Comment by jamesgorrie** [comment]
Just a bit around code clarity, but that's just opinion. 👍

File: common/app/model/Asset.scala (hunk 2 above)
Comment (sammorrisdesign) [inline common/app/model/Asset.scala]: Ha no worries. That's a good point on using the new vals for legibility.

I believe videojs needs the duration in seconds to do its magic
